ui/src/app/base/hooks.js

```diff
@@ -156,3 +156,10 @@ export const useSendAnalytics = (
     }
   }, [sendCondition, eventCategory, eventAction, eventLabel]);
 };
+
+/**
+ * Runs a function exactly once after component has undergone initial render,
+ * i.e. the component has mounted.
+ * @param {Function} func - Function to run after component mounts.
+ */
+export const useComponentDidMount = func => useEffect(func, []);
```

Comment:
Can't say I like this very much, componentDidMount is an old lifecycle event that doesn't have much meaning in the context of hooks.

useEffect(func, []) is the component is the idiomatic way to do this, I don't think we need this hook.

Reply:
@squidsoup that's fair! Originally I had it as `useHasRendered` which returned a boolean of whether or not the component has undergone initial render, but I think I generalised it a bit too far in the end.